### DISCUSSION TITLE
magit-branch-read-args should take a new parameter `default-target`

### DISCUSSION THREAD

**Discussion by nordlow** [magit/magit#5262] [Ideas]
The reason is that I want to be able to make use of a function that generates a default value for `default-target` branch for instance a function that looks like

`USER/TIME_STAMP/OPTIONAL_TAG`

where the `OPTIONAL_TAG`

doesn't need to be set as the `USER/TIME_STAMP` is always unique.

And pass the return value of the generator function to the new parameter `default-target` when needed.

**Comment by tarsius** [comment]
Merely adding a new argument to that function won't cause the commands that call it to use that argument, so I assume you intend to use that function with the new argument in some *new* "create a timestamp branch" command.

If you do that, you might as well forgo using `magit-branch-read-args` and instead merely take inspiration from that and implement your own dedicated alternative.

IMO this is too much of niche feature for it to be appropriate to implement (part of) it in Magit itself.
